AK/Math.h

```diff
@@ -60,6 +57,116 @@ INTEGER_BUILTIN(clz);
+        I res;                                                                                                                    \
+        if constexpr (sizeof(I) == sizeof(i64))                                                                                   \
+            asm(                                                                                                                  \
+                "fistpq %0"                                                                                                       \
+                : "=m"(res)                                                                                                       \
+                : "t"(value)                                                                                                      \
+                : "st");                                                                                                          \
+        if constexpr (sizeof(I) == sizeof(i32))                                                                                   \
+            asm(                                                                                                                  \
+                "fistpl %0"                                                                                                       \
+                : "=m"(res)                                                                                                       \
+                : "t"(value)                                                                                                      \
+                : "st");                                                                                                          \
+        if constexpr (sizeof(I) == sizeof(i16))                                                                                   \
+            asm(                                                                                                                  \
+                "fistps %0"                                                                                                       \
+                : "=m"(res)                                                                                                       \
+                : "t"(value)                                                                                                      \
+                : "st");                                                                                                          \
+        /* restore the controll word*/                                                                                            \
+        asm("fldcw %0" ::"m"(cw));                                                                                                \
+        return res;                                                                                                               \
+    }
+
+ROUNDING_FUNCTION(floor, FE_DOWNWARD);
```

Comment:
Can we do this without macros?

template <typename T, int roundingType> T roundFunc(...) {...}; template <FloatingPoint f> using floor = roundFunc<F, FE_DOWNWARD>;

etc?

Reply:
sure, will look at it

Reply:
I tried, bit that wont do for calling the builtins